wrapanapi/systems/openstack.py

```diff
@@ -583,15 +588,23 @@ class OpenstackSystem(System, VmMixin, TemplateMixin):
     can_suspend = True
     can_pause = True
 
-    def __init__(self, **kwargs):
-        self.keystone_version = kwargs.get('keystone_version', 2)
+    def __init__(self, tenant, username, password, auth_url, **kwargs):
+        self.keystone_version = kwargs.get("keystone_version")
+        if not self.keystone_version:
+            parsed_keystone_version = search(r'v[2-3]$', auth_url)
```

Comment:
I'm still not happy with the way you use regexps.
I'd rather you used regexp groups in correct way. 
```
parsed_keystone_version = re.search(r'v([2-3])$', auth_url)
version =  int(parsed_keystone_version.group(1))

```